registry.go

```diff
@@ -0,0 +1,114 @@
+package manet
+
+import (
+	"fmt"
+	"net"
+	"sync"
+
+	ma "github.com/jbenet/go-multiaddr"
+)
+
+type AddrParser func(a net.Addr) (ma.Multiaddr, error)
+type MaddrParser func(ma ma.Multiaddr) (net.Addr, error)
+
+var maddrParsers map[string]MaddrParser
+var addrParsers map[string]AddrParser
+var addrParsersLock sync.Mutex
```

Comment:
the reason this is better is that this allows "those who want to be careful with modularity" to be perfectly safe, while still making the defaults do the typical go expected behavior for most simple programs.